src/CMakeLists.txt

```diff
@@ -121,6 +121,7 @@ if(CMAKE_CXX_COMPILER_ID STREQUAL GNU)
   else()
     string(APPEND CMAKE_EXE_LINKER_FLAGS " -rdynamic")
   endif()
+  set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -Og")
```

Comment:
AFAIK -Og -g1 for gcc would achieve something similar to the -gline-tables-only

Reply:
@ronen-fr i wanted to add `-Og` to the existing `CMAKE_CXX_FLAGS_DEBUG`, instead of overriding it. if you believe we should also use `-g` for Clang, i'd be happy to do so.

Reply:
(I didn't know about the no CMAKE_CXX_FLAGS_DEBUG init in Clang!)

I may have misunderstood what we want to happen. But what I meant is:
if for "rel-with-debug-info" builds  (*) we only want the minimal information (enough for a backtrace), then we probably should use
clang: as you've suggested -Og -gline-tables-only
and the gcc equivalent: -Og -g1

I would assume that for a full debug builds, we'd want -Og -g for both.
 
(*) might be useful to have the traceback info in release builds, too. And in Clang, with 'line-tables-only', we won't pay for that.

Reply:
> I may have misunderstood what we want to happen. But what I meant is:

probably the commit message could provide more context.

> if for "rel-with-debug-info" builds (*) we only want the minimal information (enough for a backtrace), then we probably should use
> clang: as you've suggested -Og -gline-tables-only

i am using `-O2` for clang. as `-Og` is optimized for debugging, IMO.

> and the gcc equivalent: -Og -g1

again, i don't want to change the default setting of CMake, as it has already set some flags for `CMAKE_${lang}_FLAGS_*_INIT`. what i want is to

1. fix https://tracker.ceph.com/issues/51441
2. address the concern related to Clang. so it is at least on par with GCC when it comes to the `CMAKE_CXX_FLAGS_DEBUG`.

> 
> I would assume that for a full debug builds, we'd want -Og -g for both.

it's implemented in the latest revision.

> 
> (*) might be useful to have the traceback info in release builds, too. And in Clang, with 'line-tables-only', we won't pay for that.

Reply:
got it

Reply:
Let me check to see what `-0g -gline-tables-only` generates in regards to debugging..
And if FreeBSD gdb likes that.

I like the idea that we generate some info in the release version.
Does not have to be loaded with full symbols, but a nice backtrace would help.
(Once the backtrace problem in Clang14 is fixed.)

Reply:
'mmmm, compiling with `-Og -g` optimizes to the fact that some variables are optimized out.
Which make debugging a pain time by time.
So my preference definitly goes out to `-o0 -g`, even though resulting size of everything is bigger.
Tha is just a matter of diskspace.

Reply:
If I understand correctly, the original problem was that the resulting size was above ARM architecture limitations. So - it might be more than disk space.

Reply:
i concur with @ronen-fr . it's more about the size of the generated executable. i am not sure if we have plan to build with clang on aarch64, or if https://gcc.gnu.org/bugzilla/show_bug.cgi?id=101448 applies to clang + aarch64, so i am fine either way.

i checked the default pass of clang by comparing the output of following commands
```bash
$ echo 'int;' | clang-13 -xc -O0    - -o /dev/null -\#\#\#
$ echo 'int;' | clang-13 -xc    - -o /dev/null -\#\#\#
```
it turns out clang uses `-O0` by default.

see also https://stackoverflow.com/questions/15548023/clang-optimization-levels. so i am going to drop `-Og` from clang compile options.

Reply:
Nice trick...
[ BTW I said Clang14, but meant FreeBSD14/Clang12]
```
FreeBSD clang version 12.0.1 (git@github.com:llvm/llvm-project.git llvmorg-12.0.1-rc2-0-ge7dac564cd0e)
Target: x86_64-unknown-freebsd14.0
Thread model: posix
InstalledDir: /usr/bin
 "/usr/bin/clang" "-cc1" "-triple" "x86_64-unknown-freebsd14.0" "-emit-obj" "-mrelax-all" "--mrelax-relocations" "-disable-free" "-main-file-name" "-" "-mrelocation-model" "static" "-mframe-pointer=all" "-fno-rounding-math" "-mconstructor-aliases" "-munwind-tables" "-target-cpu" "x86-64" "-tune-cpu" "generic" "-fno-split-dwarf-inlining" "-debugger-tuning=gdb" "-resource-dir" "/usr/lib/clang/12.0.1" "-fdebug-compilation-dir" "/home/wjw/master" "-ferror-limit" "19" "-fgnuc-version=4.2.1" "-fcolor-diagnostics" "-faddrsig" "-o" "/tmp/--e30ac5.o" "-x" "c" "-"
 "/usr/local/bin/x86_64-unknown-freebsd14.0-ld" "--eh-frame-hdr" "-dynamic-linker" "/libexec/ld-elf.so.1" "--hash-style=both" "--enable-new-dtags" "-o" "/dev/null" "/usr/lib/crt1.o" "/usr/lib/crti.o" "/usr/lib/crtbegin.o" "-L/usr/lib" "/tmp/--e30ac5.o" "-lgcc" "--as-needed" "-lgcc_s" "--no-as-needed" "-lc" "-lgcc" "--as-needed" "-lgcc_s" "--no-as-needed" "/usr/lib/crtend.o" "/usr/lib/crtn.o"
```

So Clang12 does not have any default commandline settings for optimization?

Reply:
> So Clang12 does not have any default commandline settings for optimization?

that's my understanding.